<launch>
  <arg name="launch_prefix" default="" /> <!-- set to value="gdbserver localhost:10000" for remote debugging -->
  <arg name="node_namespace" default="apriltag_ros_continuous_node" />
  <arg name="camera_name" default="/camera_rect" />
  <arg name="camera_frame" default="camera" />
  <arg name="image_topic" default="image_rect" />
  <arg name="publish_tf" default="false"/>
  <arg name='settings_filename' default='settings.yaml' />
  <arg name='tags_filename' default='tags.yaml' />
  <arg name="params_package_name" default="apriltag_ros"/> <!-- package where settings.yaml and tags.yaml are located -->
  <!-- Set parameters -->
  <rosparam command="load" file="$(eval find(arg('params_package_name')) + '/config/' + arg('settings_filename'))" ns="$(arg node_namespace)" />
  <rosparam command="load" file="$(eval find(arg('params_package_name')) + '/config/' + arg('tags_filename'))" ns="$(arg node_namespace)" />

  <node pkg="apriltag_ros" type="apriltag_ros_continuous_node" name="$(arg node_namespace)" clear_params="true" output="screen" launch-prefix="$(arg launch_prefix)" >
    <!-- Remap topics from those used in code to those on the ROS network -->
    <!-- This "trick" the ROS node so that when it thinks it is subscribing to or publishing to /some_topic it is actually subscribing to or publishing to /some_other_topic, for instance.-->
    <remap from="image_rect" to="$(arg camera_name)/$(arg image_topic)" />
    <remap from="camera_info" to="$(arg camera_name)/camera_info" />
    <remap from="tag_detections" to="$(arg camera_name)/tag_detections" />
    <param name="camera_frame" type="str" value="$(arg camera_frame)" />
    <param name="publish_tag_detections_image" type="bool" value="true" />      <!-- default: false -->
    <param name="publish_tf" type="bool" value="$(arg publish_tf)"/>
  </node>
</launch>